packages/plugin-hardhat/src/utils/validations.ts

```diff
@@ -9,28 +10,42 @@ import {
   isCurrentValidationData,
 } from '@openzeppelin/upgrades-core';
 
-export async function writeValidations(hre: HardhatRuntimeEnvironment, newRunData: ValidationRunData): Promise<void> {
-  let storedData: ValidationDataCurrent | undefined;
+function lock(file: string) {
+  return lockfile.lock(file, { retries: 3, realpath: false });
```

Comment:
There are some test errors:

- If the `plugin-hardhat/cache` folder does not exist, this causes an error (which is strange since `realpath` is false):
```
[Error: ENOENT: no such file or directory, mkdir '/home/runner/work/openzeppelin-upgrades/openzeppelin-upgrades/packages/plugin-hardhat/cache/validations.json.lock'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'mkdir',
  path: '/home/runner/work/openzeppelin-upgrades/openzeppelin-upgrades/packages/plugin-hardhat/cache/validations.json.lock'
}
```

- If `plugin-hardhat/cache` exists but is empty, the tests fail at line 57 instead with `ValidationsCacheNotFound`

Reply:
With or without realpath, the containing directory needs to exist. I added mkdir before lock.